node/src/reactor/main_reactor/upgrade_shutdown.rs

```diff
@@ -49,11 +110,14 @@ impl MainReactor {
         effect_builder: EffectBuilder<MainEvent>,
         validator_weights: &EraValidatorWeights,
     ) -> UpgradeShutdownInstruction {
+        let finished_gossiping_enough = self
+            .signature_gossip_tracker
+            .finished_gossiping_enough(validator_weights);
         match self
             .storage
             .era_has_sufficient_finality_signatures(validator_weights)
```

Comment:
Can we remove this check now and rely only on `finished_gossiping_enough` since we'll only enter the `ReactorState::ShutdownForUpgrade` state once the highest block is complete.

Enforcing that all blocks in the final era are sufficiently signed could even be counterproductive (e.g. for the case where a node has just done a leap during the final era).

Reply:
I think we can. I removed it in 146b716 :+1: